app/Http/Controllers/AnswerController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Quiz;
+use App\Answer;
+use Illuminate\Http\Request;
+
+class AnswerController extends Controller
+{
+    public function store()
+    {
+        $attributes = $this->validateAnswer();
+        Answer::create([
+            'choice_id' => $attributes['answer'],
+            'lesson_id' => $attributes['lesson_id']
+        ]);
+        if($attributes['completed'] == 1) {
+            $quiz = Quiz::whereId($attributes['quiz_id'])->get()->first();
+            $quiz->completed = 1;
+            $quiz->result = $quiz->getResult($attributes['lesson_id']);
+            $quiz->save();
+        }
+
+        return redirect($attributes['next_url']);
+    }
+
+    protected function validateAnswer()
+    {
+        return request()->validate([
```

Comment:
You can use Form Request File for validation and you don't need to create a validate function

Reply:
This is fixed in c6523a90d5dce97ef45dcb15b970d517f33f88da